movies.py

```diff
@@ -0,0 +1,100 @@
+"""Movie ticket process system and database"""
+
+import pprint
+
+MOVIE_PRICE = 5.0
+
+def customer_db():
+    """Define Customer database for testing"""
+    return {
+        "Bob" : {"movies":[],
+                 "cash" : 100.0},
+        "Jim" : {"movies": ["Xmen 8: The Xmennening"],
+                 "cash" : 10.0},
+        "Cary" : {"movies": ["Gigli: The Play: The Book: The movie", "The Bromance"],
+                  "cash" : 120.0},
+        "Ricci": {"movies": [],
+                  "cash" : 4.0}
+    }
+
+def movie_db():
+    """Define movie database for testing"""
+    return {
+        "Xmen" : {"title" : "Xmen 8: The Xmennening", "seats_avaliable" : 10},
+        "Bromance" : {"title" : "The Bromance", "seats_avaliable" : 20},
+        "Gigli" : {"title" : "Gigli: The Play: The Book: The movie", "seats_avaliable" : 102}
+    }
+
+def enough_money(customer):
+    """Check if customer has adequate funds"""
+    balance = customer["cash"]
+    if balance < MOVIE_PRICE:
+        return False
+    return True
+
+def purchase_ticket(movie, customer):
+    """Performs the steps required in purchasing a ticket and returns altered customer and movie"""
+    assert enough_money(customer)
+    customer = charge(customer)
+    customer = dispense_ticket(movie, customer)
+    movie = remove_seat(movie)
+    return customer, movie
+
+def charge(customer):
+    """Charge customer the price of one ticket"""
+    customer["cash"] = customer["cash"] - MOVIE_PRICE
```

Comment:
I know it might seem redundant, but having functions which are as self-sufficient as possible can be pretty important, so I would assert that the customer has enough money in this method also. That way, let's say later you write a new feature that calls the charge() method,  you don't accidentally send people into debt. I would also add another test that captures this scenario.